CMakeLists.txt

```diff
@@ -303,6 +303,7 @@ endif ()
 
 
 if (MSVC)
+    add_definitions (-DNOMINMAX)
```

Comment:
I do most of my development on Windows, and disabling `NOMINMAX` is more or less required if you include `Windows.h` in C++. Any sane build configuration will define either this macro, or `WIN32_LEAN_AND_MEAN`. I think it is fair to expect users of the header files to ensure `min` and `max` are not defined as macros.